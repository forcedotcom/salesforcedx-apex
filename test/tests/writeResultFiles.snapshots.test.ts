/*
 * Copyright (c) 2020, salesforce.com, inc.
 * All rights reserved.
 * Licensed under the BSD 3-Clause license.
 * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
 */

import chai, { expect } from 'chai';
import { join } from 'path';
import { mkdir, readFile, rm } from 'node:fs/promises';
import { createWriteStream, existsSync } from 'node:fs';
import { tmpdir } from 'os';
import { Readable } from 'stream';
import { pipeline } from 'node:stream/promises';
// @ts-ignore - no type definitions available
import chaiJestSnapshot from 'chai-jest-snapshot';
import {
  writeResultFiles,
  ApexTestResultOutcome,
  ResultFormat,
  TestResult,
  OutputDirConfig,
  CodeCoverageResult,
  PerClassCoverage
} from '../../src';

// eslint-disable-next-line @typescript-eslint/no-namespace, @typescript-eslint/no-unused-vars
declare namespace Chai {
  interface Assertion {
    matchSnapshot(): Assertion;
  }
}

// Configure chai to use jest snapshots
chai.use(chaiJestSnapshot);

describe('writeResultFiles - Snapshot Tests', () => {
  let tempDir: string;

  before(() => {
    chaiJestSnapshot.resetSnapshotRegistry();
  });

  beforeEach(async function () {
    chaiJestSnapshot.configureUsingMochaContext(this);
    // Configure snapshots to be saved in the source directory, not lib
    chaiJestSnapshot.setFilename(
      join(
        __dirname,
        '__snapshots__',
        'writeResultFiles.snapshots.test.ts.snap'
      )
    );
    tempDir = join(
      tmpdir(),
      `apex-snapshot-test-${Date.now()}-${Math.random().toString(36).substr(2, 9)}`
    );
    await mkdir(tempDir, { recursive: true });
  });

  afterEach(async () => {
    if (existsSync(tempDir)) {
      await rm(tempDir, { recursive: true, force: true });
    }
  });

  // Mock runPipeline function for testing
  const mockRunPipeline = async (
    readable: Readable,
    filePath: string
  ): Promise<string> => {
    await pipeline(readable, createWriteStream(filePath));
    return filePath;
  };

  const mockTestResult: TestResult = {
    summary: {
      outcome: 'Completed',
      testsRan: 3,
      passing: 2,
      failing: 1,
      skipped: 0,
      passRate: '67%',
      failRate: '33%',
      skipRate: '0%',
      testStartTime: '2023-01-01T12:00:00.000Z',
      testExecutionTimeInMs: 1500,
      testTotalTimeInMs: 2500,
      commandTimeInMs: 3500,
      hostname: 'snapshot-test-host',
      orgId: '00D000000000000EAA',
      username: 'snapshot@example.com',
      testRunId: 'snapshot-test-run-456',
      userId: '005000000000000AAA'
    },
    tests: [
      {
        id: '01p000000000001AAA',
        queueItemId: '709000000000001AAA',
        stackTrace: null as string | null,
        message: null as string | null,
        asyncApexJobId: '707000000000001AAA',
        methodName: 'testSnapshotMethod1',
        outcome: ApexTestResultOutcome.Pass,
        apexLogId: null as string | null,
        apexClass: {
          id: '01p000000000001AAA',
          name: 'SnapshotTestClass1',
          namespacePrefix: null as string | null,
          fullName: 'SnapshotTestClass1'
        },
        runTime: 750,
        testTimestamp: '2023-01-01T12:00:00.500Z',
        fullName: 'SnapshotTestClass1.testSnapshotMethod1'
      },
      {
        id: '01p000000000002AAA',
        queueItemId: '709000000000002AAA',
        stackTrace: 'System.AssertException: Snapshot test assertion failed',
        message: 'Snapshot test failed with detailed error message',
        asyncApexJobId: '707000000000001AAA',
        methodName: 'testSnapshotMethod2',
        outcome: ApexTestResultOutcome.Fail,
        apexLogId: '07L000000000001AAA',
        apexClass: {
          id: '01p000000000002AAA',
          name: 'SnapshotTestClass2',
          namespacePrefix: 'TestNamespace',
          fullName: 'TestNamespace.SnapshotTestClass2'
        },
        runTime: 450,
        testTimestamp: '2023-01-01T12:00:01.200Z',
        fullName: 'TestNamespace.SnapshotTestClass2.testSnapshotMethod2'
      },
      {
        id: '01p000000000003AAA',
        queueItemId: '709000000000003AAA',
        stackTrace: null as string | null,
        message: null as string | null,
        asyncApexJobId: '707000000000001AAA',
        methodName: 'testSnapshotMethod3',
        outcome: ApexTestResultOutcome.Pass,
        apexLogId: null as string | null,
        apexClass: {
          id: '01p000000000003AAA',
          name: 'SnapshotTestClass3',
          namespacePrefix: null as string | null,
          fullName: 'SnapshotTestClass3'
        },
        runTime: 300,
        testTimestamp: '2023-01-01T12:00:01.800Z',
        fullName: 'SnapshotTestClass3.testSnapshotMethod3'
      }
    ]
  };

  const mockCodeCoverageResult: CodeCoverageResult = {
    apexId: '01p000000000001AAA',
    name: 'SnapshotTestClass1',
    type: 'ApexClass' as const,
    numLinesCovered: 8,
    numLinesUncovered: 2,
    percentage: '80%',
    coveredLines: [1, 2, 3, 4, 5, 6, 7, 8],
    uncoveredLines: [9, 10]
  };

  const mockPerClassCoverage: PerClassCoverage = {
    apexClassOrTriggerName: 'SnapshotTestClass1',
    apexClassOrTriggerId: '01p000000000001AAA',
    apexTestClassId: '01p000000000002AAA',
    apexTestMethodName: 'testSnapshotMethod1',
    numLinesCovered: 8,
    numLinesUncovered: 2,
    percentage: '80%',
    coverage: mockCodeCoverageResult
  };

  it('should produce consistent JSON output', async function () {
    const outputConfig: OutputDirConfig = {
      dirPath: tempDir,
      resultFormats: [ResultFormat.json]
    };

    await writeResultFiles(
      mockTestResult,
      outputConfig,
      false,
      mockRunPipeline
    );

    const jsonFilePath = join(
      tempDir,
      'test-result-snapshot-test-run-456.json'
    );
    const content = await readFile(jsonFilePath, 'utf8');
    const parsedContent = JSON.parse(content);

    // Snapshot the parsed JSON structure
    expect(parsedContent).to.matchSnapshot();
  });

  it('should produce consistent TAP output', async function () {
    const outputConfig: OutputDirConfig = {
      dirPath: tempDir,
      resultFormats: [ResultFormat.tap]
    };

    await writeResultFiles(
      mockTestResult,
      outputConfig,
      false,
      mockRunPipeline
    );

    const tapFilePath = join(
      tempDir,
      'test-result-snapshot-test-run-456-tap.txt'
    );
    const content = await readFile(tapFilePath, 'utf8');

    // Snapshot the TAP output
    expect(content).to.matchSnapshot();
  });

  it('should produce consistent JUnit output', async function () {
    const outputConfig: OutputDirConfig = {
      dirPath: tempDir,
      resultFormats: [ResultFormat.junit]
    };

    await writeResultFiles(
      mockTestResult,
      outputConfig,
      false,
      mockRunPipeline
    );

    const junitFilePath = join(
      tempDir,
      'test-result-snapshot-test-run-456-junit.xml'
    );
    const content = await readFile(junitFilePath, 'utf8');

    // Snapshot the JUnit XML output
    expect(content).to.matchSnapshot();
  });

  it('should produce consistent code coverage output', async function () {
    const testResultWithCoverage: TestResult = {
      ...mockTestResult,
      tests: [
        {
          ...mockTestResult.tests[0],
          perClassCoverage: [mockPerClassCoverage]
        },
        ...mockTestResult.tests.slice(1)
      ]
    };

    const outputConfig: OutputDirConfig = {
      dirPath: tempDir
    };

    await writeResultFiles(
      testResultWithCoverage,
      outputConfig,
      true,
      mockRunPipeline
    );

    const coverageFilePath = join(
      tempDir,
      'test-result-snapshot-test-run-456-codecoverage.json'
    );
    const content = await readFile(coverageFilePath, 'utf8');
    const parsedContent = JSON.parse(content);

    // Snapshot the code coverage structure
    expect(parsedContent).to.matchSnapshot();
  });

  it('should produce consistent fileInfos output', async function () {
    const outputConfig: OutputDirConfig = {
      dirPath: tempDir,
      fileInfos: [
        {
          filename: 'snapshot-custom.txt',
          content: 'Snapshot custom content for testing'
        },
        {
          filename: 'snapshot-metadata.json',
          content: {
            timestamp: '2023-01-01T12:00:00.000Z',
            version: '2.0.0',
            environment: 'snapshot-test',
            features: ['feature1', 'feature2'],
            config: {
              debug: true,
              timeout: 5000,
              retries: 3
            }
          }
        }
      ]
    };

    await writeResultFiles(
      mockTestResult,
      outputConfig,
      false,
      mockRunPipeline
    );

    // Test string content
    const textFilePath = join(tempDir, 'snapshot-custom.txt');
    const textContent = await readFile(textFilePath, 'utf8');
    expect(textContent).to.matchSnapshot();

    // Test JSON object content
    const jsonFilePath = join(tempDir, 'snapshot-metadata.json');
    const jsonContent = await readFile(jsonFilePath, 'utf8');
    const parsedJsonContent = JSON.parse(jsonContent);
    expect(parsedJsonContent).to.matchSnapshot();
  });

  it('should produce consistent comprehensive output with all options', async function () {
    const testResultWithCoverage: TestResult = {
      ...mockTestResult,
      tests: [
        {
          ...mockTestResult.tests[0],
          perClassCoverage: [mockPerClassCoverage]
        },
        ...mockTestResult.tests.slice(1)
      ]
    };

    const outputConfig: OutputDirConfig = {
      dirPath: tempDir,
      resultFormats: [ResultFormat.json, ResultFormat.tap, ResultFormat.junit],
      fileInfos: [
        {
          filename: 'comprehensive-test.txt',
          content: 'Comprehensive test content'
        },
        {
          filename: 'comprehensive-config.json',
          content: {
            test: 'comprehensive',
            settings: {
              parallel: true,
              verbose: false
            }
          }
        }
      ]
    };

    const result = await writeResultFiles(
      testResultWithCoverage,
      outputConfig,
      true,
      mockRunPipeline
    );

    // Snapshot the list of created files (sorted for consistency)
    const sortedResult = result
      .map((path) => path.replace(tempDir, '[TEMP_DIR]'))
      .sort();
    expect(sortedResult).to.matchSnapshot();

    // Snapshot each file's content
    for (const filePath of result) {
      const fileName = filePath.split('/').pop();
      const content = await readFile(filePath, 'utf8');

      if (fileName?.endsWith('.json')) {
        const parsedContent = JSON.parse(content);
        expect(parsedContent).to.matchSnapshot();
      } else {
        expect(content).to.matchSnapshot();
      }
    }
  });
});
