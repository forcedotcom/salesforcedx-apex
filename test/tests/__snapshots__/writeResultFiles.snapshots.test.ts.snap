// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`writeResultFiles - Snapshot Tests should produce consistent JSON output 1`] = `
Object {
  "summary": Object {
    "commandTimeInMs": 3500,
    "failRate": "33%",
    "failing": 1,
    "hostname": "snapshot-test-host",
    "orgId": "00D000000000000EAA",
    "outcome": "Completed",
    "passRate": "67%",
    "passing": 2,
    "skipRate": "0%",
    "skipped": 0,
    "testExecutionTimeInMs": 1500,
    "testRunId": "snapshot-test-run-456",
    "testStartTime": "2023-01-01T12:00:00.000Z",
    "testTotalTimeInMs": 2500,
    "testsRan": 3,
    "userId": "005000000000000AAA",
    "username": "snapshot@example.com",
  },
  "tests": Array [
    Object {
      "apexClass": Object {
        "fullName": "SnapshotTestClass1",
        "id": "01p000000000001AAA",
        "name": "SnapshotTestClass1",
        "namespacePrefix": null,
      },
      "apexLogId": null,
      "asyncApexJobId": "707000000000001AAA",
      "fullName": "SnapshotTestClass1.testSnapshotMethod1",
      "id": "01p000000000001AAA",
      "message": null,
      "methodName": "testSnapshotMethod1",
      "outcome": "Pass",
      "queueItemId": "709000000000001AAA",
      "runTime": 750,
      "stackTrace": null,
      "testTimestamp": "2023-01-01T12:00:00.500Z",
    },
    Object {
      "apexClass": Object {
        "fullName": "TestNamespace.SnapshotTestClass2",
        "id": "01p000000000002AAA",
        "name": "SnapshotTestClass2",
        "namespacePrefix": "TestNamespace",
      },
      "apexLogId": "07L000000000001AAA",
      "asyncApexJobId": "707000000000001AAA",
      "fullName": "TestNamespace.SnapshotTestClass2.testSnapshotMethod2",
      "id": "01p000000000002AAA",
      "message": "Snapshot test failed with detailed error message",
      "methodName": "testSnapshotMethod2",
      "outcome": "Fail",
      "queueItemId": "709000000000002AAA",
      "runTime": 450,
      "stackTrace": "System.AssertException: Snapshot test assertion failed",
      "testTimestamp": "2023-01-01T12:00:01.200Z",
    },
    Object {
      "apexClass": Object {
        "fullName": "SnapshotTestClass3",
        "id": "01p000000000003AAA",
        "name": "SnapshotTestClass3",
        "namespacePrefix": null,
      },
      "apexLogId": null,
      "asyncApexJobId": "707000000000001AAA",
      "fullName": "SnapshotTestClass3.testSnapshotMethod3",
      "id": "01p000000000003AAA",
      "message": null,
      "methodName": "testSnapshotMethod3",
      "outcome": "Pass",
      "queueItemId": "709000000000003AAA",
      "runTime": 300,
      "stackTrace": null,
      "testTimestamp": "2023-01-01T12:00:01.800Z",
    },
  ],
}
`;

exports[`writeResultFiles - Snapshot Tests should produce consistent JUnit output 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<testsuites>
    <testsuite name=\\"force.apex\\" timestamp=\\"2023-01-01T12:00:00.000Z\\" hostname=\\"snapshot-test-host\\" tests=\\"3\\" failures=\\"1\\"  errors=\\"0\\"  time=\\"1.50\\">
        <properties>
            <property name=\\"outcome\\" value=\\"Completed\\"/>
            <property name=\\"testsRan\\" value=\\"3\\"/>
            <property name=\\"passing\\" value=\\"2\\"/>
            <property name=\\"failing\\" value=\\"1\\"/>
            <property name=\\"skipped\\" value=\\"0\\"/>
            <property name=\\"passRate\\" value=\\"67%\\"/>
            <property name=\\"failRate\\" value=\\"33%\\"/>
            <property name=\\"testStartTime\\" value=\\"Sun Jan 01 2023 9:00:00 AM\\"/>
            <property name=\\"testExecutionTime\\" value=\\"1.50 s\\"/>
            <property name=\\"testTotalTime\\" value=\\"2.50 s\\"/>
            <property name=\\"commandTime\\" value=\\"3.50 s\\"/>
            <property name=\\"hostname\\" value=\\"snapshot-test-host\\"/>
            <property name=\\"orgId\\" value=\\"00D000000000000EAA\\"/>
            <property name=\\"username\\" value=\\"snapshot@example.com\\"/>
            <property name=\\"testRunId\\" value=\\"snapshot-test-run-456\\"/>
            <property name=\\"userId\\" value=\\"005000000000000AAA\\"/>
        </properties>
        <testcase name=\\"testSnapshotMethod1\\" classname=\\"SnapshotTestClass1\\" time=\\"0.75\\">
        </testcase>
        <testcase name=\\"testSnapshotMethod2\\" classname=\\"TestNamespace.SnapshotTestClass2\\" time=\\"0.45\\">
            <failure message=\\"Snapshot test failed with detailed error message\\"><![CDATA[System.AssertException: Snapshot test assertion failed]]></failure>
        </testcase>
        <testcase name=\\"testSnapshotMethod3\\" classname=\\"SnapshotTestClass3\\" time=\\"0.30\\">
        </testcase>
    </testsuite>
</testsuites>
"
`;

exports[`writeResultFiles - Snapshot Tests should produce consistent TAP output 1`] = `
"1..3
ok 1 SnapshotTestClass1.testSnapshotMethod1
not ok 2 TestNamespace.SnapshotTestClass2.testSnapshotMethod2
# Snapshot test failed with detailed error message
# System.AssertException: Snapshot test assertion failed
ok 3 SnapshotTestClass3.testSnapshotMethod3
"
`;

exports[`writeResultFiles - Snapshot Tests should produce consistent code coverage output 1`] = `
Array [
  Array [
    Object {
      "apexClassOrTriggerId": "01p000000000001AAA",
      "apexClassOrTriggerName": "SnapshotTestClass1",
      "apexTestClassId": "01p000000000002AAA",
      "apexTestMethodName": "testSnapshotMethod1",
      "coverage": Object {
        "apexId": "01p000000000001AAA",
        "coveredLines": Array [
          1,
          2,
          3,
          4,
          5,
          6,
          7,
          8,
        ],
        "name": "SnapshotTestClass1",
        "numLinesCovered": 8,
        "numLinesUncovered": 2,
        "percentage": "80%",
        "type": "ApexClass",
        "uncoveredLines": Array [
          9,
          10,
        ],
      },
      "numLinesCovered": 8,
      "numLinesUncovered": 2,
      "percentage": "80%",
    },
  ],
]
`;

exports[`writeResultFiles - Snapshot Tests should produce consistent comprehensive output with all options 1`] = `
Array [
  "[TEMP_DIR]/comprehensive-config.json",
  "[TEMP_DIR]/comprehensive-test.txt",
  "[TEMP_DIR]/test-result-snapshot-test-run-456-codecoverage.json",
  "[TEMP_DIR]/test-result-snapshot-test-run-456-junit.xml",
  "[TEMP_DIR]/test-result-snapshot-test-run-456-tap.txt",
  "[TEMP_DIR]/test-result-snapshot-test-run-456.json",
  "[TEMP_DIR]/test-run-id.txt",
]
`;

exports[`writeResultFiles - Snapshot Tests should produce consistent comprehensive output with all options 2`] = `"snapshot-test-run-456"`;

exports[`writeResultFiles - Snapshot Tests should produce consistent comprehensive output with all options 3`] = `
Object {
  "summary": Object {
    "commandTimeInMs": 3500,
    "failRate": "33%",
    "failing": 1,
    "hostname": "snapshot-test-host",
    "orgId": "00D000000000000EAA",
    "outcome": "Completed",
    "passRate": "67%",
    "passing": 2,
    "skipRate": "0%",
    "skipped": 0,
    "testExecutionTimeInMs": 1500,
    "testRunId": "snapshot-test-run-456",
    "testStartTime": "2023-01-01T12:00:00.000Z",
    "testTotalTimeInMs": 2500,
    "testsRan": 3,
    "userId": "005000000000000AAA",
    "username": "snapshot@example.com",
  },
  "tests": Array [
    Object {
      "apexClass": Object {
        "fullName": "SnapshotTestClass1",
        "id": "01p000000000001AAA",
        "name": "SnapshotTestClass1",
        "namespacePrefix": null,
      },
      "apexLogId": null,
      "asyncApexJobId": "707000000000001AAA",
      "fullName": "SnapshotTestClass1.testSnapshotMethod1",
      "id": "01p000000000001AAA",
      "message": null,
      "methodName": "testSnapshotMethod1",
      "outcome": "Pass",
      "perClassCoverage": Array [
        Object {
          "apexClassOrTriggerId": "01p000000000001AAA",
          "apexClassOrTriggerName": "SnapshotTestClass1",
          "apexTestClassId": "01p000000000002AAA",
          "apexTestMethodName": "testSnapshotMethod1",
          "coverage": Object {
            "apexId": "01p000000000001AAA",
            "coveredLines": Array [
              1,
              2,
              3,
              4,
              5,
              6,
              7,
              8,
            ],
            "name": "SnapshotTestClass1",
            "numLinesCovered": 8,
            "numLinesUncovered": 2,
            "percentage": "80%",
            "type": "ApexClass",
            "uncoveredLines": Array [
              9,
              10,
            ],
          },
          "numLinesCovered": 8,
          "numLinesUncovered": 2,
          "percentage": "80%",
        },
      ],
      "queueItemId": "709000000000001AAA",
      "runTime": 750,
      "stackTrace": null,
      "testTimestamp": "2023-01-01T12:00:00.500Z",
    },
    Object {
      "apexClass": Object {
        "fullName": "TestNamespace.SnapshotTestClass2",
        "id": "01p000000000002AAA",
        "name": "SnapshotTestClass2",
        "namespacePrefix": "TestNamespace",
      },
      "apexLogId": "07L000000000001AAA",
      "asyncApexJobId": "707000000000001AAA",
      "fullName": "TestNamespace.SnapshotTestClass2.testSnapshotMethod2",
      "id": "01p000000000002AAA",
      "message": "Snapshot test failed with detailed error message",
      "methodName": "testSnapshotMethod2",
      "outcome": "Fail",
      "queueItemId": "709000000000002AAA",
      "runTime": 450,
      "stackTrace": "System.AssertException: Snapshot test assertion failed",
      "testTimestamp": "2023-01-01T12:00:01.200Z",
    },
    Object {
      "apexClass": Object {
        "fullName": "SnapshotTestClass3",
        "id": "01p000000000003AAA",
        "name": "SnapshotTestClass3",
        "namespacePrefix": null,
      },
      "apexLogId": null,
      "asyncApexJobId": "707000000000001AAA",
      "fullName": "SnapshotTestClass3.testSnapshotMethod3",
      "id": "01p000000000003AAA",
      "message": null,
      "methodName": "testSnapshotMethod3",
      "outcome": "Pass",
      "queueItemId": "709000000000003AAA",
      "runTime": 300,
      "stackTrace": null,
      "testTimestamp": "2023-01-01T12:00:01.800Z",
    },
  ],
}
`;

exports[`writeResultFiles - Snapshot Tests should produce consistent comprehensive output with all options 4`] = `
"1..3
ok 1 SnapshotTestClass1.testSnapshotMethod1
not ok 2 TestNamespace.SnapshotTestClass2.testSnapshotMethod2
# Snapshot test failed with detailed error message
# System.AssertException: Snapshot test assertion failed
ok 3 SnapshotTestClass3.testSnapshotMethod3
"
`;

exports[`writeResultFiles - Snapshot Tests should produce consistent comprehensive output with all options 5`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<testsuites>
    <testsuite name=\\"force.apex\\" timestamp=\\"2023-01-01T12:00:00.000Z\\" hostname=\\"snapshot-test-host\\" tests=\\"3\\" failures=\\"1\\"  errors=\\"0\\"  time=\\"1.50\\">
        <properties>
            <property name=\\"outcome\\" value=\\"Completed\\"/>
            <property name=\\"testsRan\\" value=\\"3\\"/>
            <property name=\\"passing\\" value=\\"2\\"/>
            <property name=\\"failing\\" value=\\"1\\"/>
            <property name=\\"skipped\\" value=\\"0\\"/>
            <property name=\\"passRate\\" value=\\"67%\\"/>
            <property name=\\"failRate\\" value=\\"33%\\"/>
            <property name=\\"testStartTime\\" value=\\"Sun Jan 01 2023 9:00:00 AM\\"/>
            <property name=\\"testExecutionTime\\" value=\\"1.50 s\\"/>
            <property name=\\"testTotalTime\\" value=\\"2.50 s\\"/>
            <property name=\\"commandTime\\" value=\\"3.50 s\\"/>
            <property name=\\"hostname\\" value=\\"snapshot-test-host\\"/>
            <property name=\\"orgId\\" value=\\"00D000000000000EAA\\"/>
            <property name=\\"username\\" value=\\"snapshot@example.com\\"/>
            <property name=\\"testRunId\\" value=\\"snapshot-test-run-456\\"/>
            <property name=\\"userId\\" value=\\"005000000000000AAA\\"/>
        </properties>
        <testcase name=\\"testSnapshotMethod1\\" classname=\\"SnapshotTestClass1\\" time=\\"0.75\\">
        </testcase>
        <testcase name=\\"testSnapshotMethod2\\" classname=\\"TestNamespace.SnapshotTestClass2\\" time=\\"0.45\\">
            <failure message=\\"Snapshot test failed with detailed error message\\"><![CDATA[System.AssertException: Snapshot test assertion failed]]></failure>
        </testcase>
        <testcase name=\\"testSnapshotMethod3\\" classname=\\"SnapshotTestClass3\\" time=\\"0.30\\">
        </testcase>
    </testsuite>
</testsuites>
"
`;

exports[`writeResultFiles - Snapshot Tests should produce consistent comprehensive output with all options 6`] = `
Array [
  Array [
    Object {
      "apexClassOrTriggerId": "01p000000000001AAA",
      "apexClassOrTriggerName": "SnapshotTestClass1",
      "apexTestClassId": "01p000000000002AAA",
      "apexTestMethodName": "testSnapshotMethod1",
      "coverage": Object {
        "apexId": "01p000000000001AAA",
        "coveredLines": Array [
          1,
          2,
          3,
          4,
          5,
          6,
          7,
          8,
        ],
        "name": "SnapshotTestClass1",
        "numLinesCovered": 8,
        "numLinesUncovered": 2,
        "percentage": "80%",
        "type": "ApexClass",
        "uncoveredLines": Array [
          9,
          10,
        ],
      },
      "numLinesCovered": 8,
      "numLinesUncovered": 2,
      "percentage": "80%",
    },
  ],
]
`;

exports[`writeResultFiles - Snapshot Tests should produce consistent comprehensive output with all options 7`] = `"Comprehensive test content"`;

exports[`writeResultFiles - Snapshot Tests should produce consistent comprehensive output with all options 8`] = `
Object {
  "settings": Object {
    "parallel": true,
    "verbose": false,
  },
  "test": "comprehensive",
}
`;

exports[`writeResultFiles - Snapshot Tests should produce consistent fileInfos output 1`] = `"Snapshot custom content for testing"`;

exports[`writeResultFiles - Snapshot Tests should produce consistent fileInfos output 2`] = `
Object {
  "config": Object {
    "debug": true,
    "retries": 3,
    "timeout": 5000,
  },
  "environment": "snapshot-test",
  "features": Array [
    "feature1",
    "feature2",
  ],
  "timestamp": "2023-01-01T12:00:00.000Z",
  "version": "2.0.0",
}
`;
